src/components/UploadingJobsDisplay/UploadingJobsDisplay.js

```diff
@@ -48,6 +48,29 @@ class UploadingJobsDisplay extends Component {
     this.uploadJobs();
   }
 
+  componentDidUpdate() {
+    if (!this.props.files) {
+      return;
+    }
+
+    let dontLeave = false;
+
+    for (const file in this.props.files) {
```

Comment:
Let's use the code below instead of `for in` loop.
```suggestion
    const isAbleToLeavePage = Object.keys(this.props.files)
      .some(file => file.status === fileStatuses.UPLOADING);
```

Reply:
Thanks, refactored